source/scheduler.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <queue>
+
+#include "event.hpp"
+
+namespace sim {
+
+// Scheduler is implemented as a Singleton class
+// which provides a global access to a single instance
+class Scheduler {
+public:
+    // Static method to get the instance
+    static Scheduler& get_instance() {
+        static Scheduler instance;
+        return instance;
+    }
+
+    void tick();
+    void add(Event event);
```

Comment:
The Event is abstract, so we need to pass it by reference or pass some kind of pointer to the Event
```suggestion
    void add(Event& event);
```